storage/remote/client.go

```diff
@@ -99,7 +100,7 @@ type ClientConfig struct {
 // ReadClient uses the SAMPLES method of remote read to read series samples from remote server.
 // TODO(bwplotka): Add streamed chunked remote read method as well (https://github.com/prometheus/prometheus/issues/5926).
 type ReadClient interface {
-	Read(ctx context.Context, query *prompb.Query) (*prompb.QueryResult, error)
+	StartRead(ctx context.Context, query *prompb.Query, sortSeries bool) (storage.SeriesSet, error)
```

Comment:
Do we need to rename the method?

Reply:
I think `StartRead` is asynchronous method name,  it imply that the result `storage.SeriesSet` must be handled.

Reply:
We do this all the time, we have method Query and `Select` that gives SeriesSet so it might be ok to have just `Read` for simplicity and consistency (:

Reply:
Fixed.